server/matchmaker/search.py

```diff
@@ -16,6 +17,28 @@
 OnMatchedCallback = Callable[["Search", "Search"], Any]
 
 
+class Game:
+    """
+    Represents a potential game
+    """
+    def __init__(
+        self,
+        match: Match,
+        match_quality: float
+    ):
```

Comment:
I would use a `typing.NamedTuple` as long as having immutable objects is OK, and if not then maybe you want a `dataclass`.